main.py

```diff
@@ -1,59 +1,105 @@
+    "ipv6": "https://www-public.imtbs-tsp.eu/~maigron/rir-stats/rir-delegations/delegations/ipv6/{country}-ipv6-delegations.html"
+}
+
+def fetch_data(country_code, data_type):
+    """Fetch ASN, IPv4, or IPv6 data for a given country code."""
+    url = BASE_URLS[data_type].format(country=country_code.lower())
+    try:
+        response = requests.get(url, headers=HEADERS)
+        response.raise_for_status()
+        soup = BeautifulSoup(response.text, "lxml")
+
+        # Locate the table
+        table = soup.find("table", attrs={"class": f"delegs {data_type} ripencc"})
+        if not table:
+            console.log(f"[yellow]No data table found for {data_type.upper()} in {country_code}.[/yellow]")
+            return country_code, data_type, None, None
+
+        # Extract headers and rows
+        headers = [header.text.strip() for header in table.find_all("th")]
+        rows = table.find_all("tr")[2:] 
+
+        # Collect data rows
+        data_rows = []
+        allocations = []
+        for row in rows:
+            columns = [td.text.strip() for td in row.find_all("td")]
+            if columns:
+                row_data = dict(zip(headers[1:], columns)) 
+                data_rows.append(row_data)
+                
+                if data_type == "asn" and columns[6] == "Allocated":
+                    allocations.append(columns[3])  # Collect allocateds
+                elif data_type in ["ipv4", "ipv6"] and columns[7] == "Allocated":
+                    ip_with_prefix = f"{columns[3]}{columns[4].strip()}"
+                    allocations.append(ip_with_prefix)
+
+        return country_code, data_type, data_rows, allocations
+
+    except requests.exceptions.RequestException as e:
+        console.log(f"[red]Error fetching {data_type.upper()} data for {country_code}: {e}[/red]")
+        return country_code, data_type, None, None
+
+# Run fetch requests in parallel
+country_data = {}
+output_dir = "output_data"
+os.makedirs(output_dir, exist_ok=True)
+
+console.log("\t[blue]Fetching data for countries...[/blue]")
+
+# Prepare to fetch all specified data types
+data_types = [args.data_type] if args.data_type != "all" else ["asn", "ipv4", "ipv6"]
+
+with ThreadPoolExecutor() as executor:
+    futures = [executor.submit(fetch_data, country, data_type) 
+               for country in args.countries 
+               for data_type in data_types]
+
+    for future in track(as_completed(futures), total=len(futures), description="Processing data..."):
+        country_code, data_type, data_rows, allocations = future.result()
+        if data_rows is not None:
+
+            # Save data to CSV for each country and type
+            df = pd.DataFrame(data_rows)
+            csv_filename = os.path.join(output_dir, f"{country_code}_{data_type}_list.csv")
+            df.to_csv(csv_filename, index=False)
+            console.log(f"[green]Data saved for {data_type.upper()} in {country_code}[/green]")
```

Comment:
Using `console.log()` in track is not a best practice. But in this case that this log will make it more readable, you should use a new console instance with new settings:

```python
console_no_time = Console(log_path=False, log_time=False)
```

and starting space for improving readability

```python
console_no_time.log(f" [green]Data saved...
```